src/backend/executor/nodeShareInputScan.c

```diff
@@ -552,11 +551,59 @@ static void sisc_lockname(char* p, int size, int share_id, const char* name)
 	}
 }
 
+
+char *joint_lock_file_name(ShareInput_Lk_Context *lk_ctxt, char *name)
+{
+	char *lock_file = palloc0(sizeof(char) * MAXPGPATH);
+
+	if(strncmp("writer", name, 6) ==0 )
+	{
+		strncat(lock_file, lk_ctxt->lkname_ready, strlen(lk_ctxt->lkname_ready));
```

Comment:
The strncat usage is not correct. It should be
		strncat(lock_file, lk_ctxt->lkname_ready, MAXPGPATH - strlen(lock_file) - 1);

From the manpage.
     strncat(char *restrict s1, const char *restrict s2, size_t n);
     The strncat() function appends not more than n characters from s2, and then adds a terminating `\0'.